lib/taskana-core/src/test/java/pro/taskana/security/JaasExtensionTest.java

```diff
@@ -58,33 +80,649 @@ void should_NotSetJaasSubject_When_MultipleAnnotationsExist_On_BeforeEach() {
     assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
   }
 
+  // endregion
+
+  // region JaasExtension#interceptAfterEachMethod
+
+  @AfterEach
+  void should_NotSetJaasSubject_When_AnnotationIsMissing_On_AfterEach() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
+  }
+
+  @WithAccessId(user = "aftereach")
+  @AfterEach
+  void should_SetJaasSubject_When_AnnotationExists_On_AfterEach() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo("aftereach");
+  }
+
+  @WithAccessId(user = "aftereach")
+  @WithAccessId(user = "afterach2")
+  @AfterEach
+  void should_NotSetJaasSubject_When_MultipleAnnotationsExist_On_AfterEach() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
+  }
+
+  // endregion
+
+  // region JaasExtension#interceptAfterAllMethod
+
+  @AfterAll
+  static void should_NotSetJaasSubject_When_AnnotationIsMissing_On_AfterAll() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
+  }
+
+  @WithAccessId(user = "afterall")
+  @AfterAll
+  static void should_SetJaasSubject_When_AnnotationExists_On_AfterAll() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo("afterall");
+  }
+
+  @WithAccessId(user = "afterall")
+  @WithAccessId(user = "afterall2")
+  @AfterAll
+  static void should_NotSetJaasSubject_When_MultipleAnnotationsExist_On_AfterAll() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
+  }
+
+  // endregion
+
+  // region JaasExtension#interceptTestMethod
+
+  @Test
+  void should_NotSetJaasSubject_When_AnnotationIsMissing_On_Test() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo(null);
+  }
+
   @WithAccessId(user = "user")
   @Test
   void should_SetJaasSubject_When_AnnotationExists_On_Test() {
     assertThat(CurrentUserContext.getUserid()).isEqualTo("user");
+    assertThat(CurrentUserContext.getGroupIds()).isEmpty();
+  }
+
+  @WithAccessId(
+      user = "user",
+      groups = {"hans", "peter"})
+  @Test
+  void should_SetJaasSubjectWithGroups_When_AnnotationExistsWithGroups_On_Test() {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo("user");
+    assertThat(CurrentUserContext.getGroupIds()).containsOnly("hans", "peter");
+  }
+
+  @WithAccessId(user = "user")
+  @Test
+  @Disabled("this can be tested with a org.junit.platform.launcher.TestExecutionListener")
+  void should_NotInjectParameter_When_ParameterIsPresent_On_Test(WithAccessId accessId) {
+    assertThat(CurrentUserContext.getUserid()).isEqualTo("user");
   }
 
   @WithAccessId(user = "user")
   @WithAccessId(user = "user2")
   @Test
-  @Disabled("can we make this work somehow?")
-  void should_NotSetJaasSubject_When_MultipleAnnotationsExist_On_Test() {
+  @Disabled("this can be tested with a org.junit.platform.launcher.TestExecutionListener")
+  void should_ThrowException_When_MultipleAnnotationsExist_On_Test() {
```

Comment:
So this test has to be extracted from this class, because the target of TestExecutionListeners annotation is ElementType.Type?